cupy/fft/fft.py

```diff
@@ -83,6 +83,7 @@ def _exec_fft(a, direction, value_type, norm, axis, overwrite_x,
         out_size = a.shape[-1]
 
     batch = a.size // a.shape[-1]
+    plan = cufft.get_current_plan()
```

Comment:
Doesn't this cause `_exec_fft` to ignore any `plan` the user passed in as a `kwarg`? i.e. if plan is not None, and we are not in a plan context then the provided plan would not be used and Plan1d would be called?

I thought the logic in this section would be something like:

```Python
if plan is None:
    plan = cufft.get_current_plan()
    if plan is None:
        plan = cufft.Plan1d(out_size, fft_type, batch)
else:
```

Reply:
or alternatively, like in the `_default_fft_func` below:
```Python
curr_plan = cufft.get_current_plan()
if curr_plan is not None:
    plan = curr_plan
```

Reply:
Yeah good catch! Thank you, @grlee77. I probably got distracted when testing and forgot to revisit this part...

For the record, I was thinking the precedence is: 
1. plan from context
2. plan from argument
3. auto planning (for `cupy.fft`). 

So, if the user uses both context and argument, by design I should have raised an error. I'll add this check when fixing the overridden plan.